cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -250,7 +258,12 @@ public void getAllApps(HttpRequest request, HttpResponder responder,
       .map(ApplicationRecord::new)
       .collect(Collectors.toList());
 
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(apps));
+    if (Optional.ofNullable(pageSize).orElse(0) != 0) {
+      PaginatedApplicationRecords result = new PaginatedApplicationRecords(apps, "sampleToken");
```

Comment:
This way you'll get infinite list or repeated apps. Probably it's better to start with null token or return an empty list if token is not empty (this way you'll have two pages: first real one and then the empty one